app/perfsymboltable.cpp

```diff
@@ -330,12 +331,12 @@ int PerfSymbolTable::insertSubprogram(CuDieRangeMapping *cudie, Dwarf_Die *top,
     int locationId = m_unwind->resolveLocation(PerfUnwind::Location(entry, fileId, m_pid, line,
                                                                     column, inlineCallLocationId));
     qint32 symId = m_unwind->resolveString(cudie->dieName(top));
```

Comment:
cudie->dieName now needs to return a struct that holds both, the mangled and the demangled name, no and then you need to pass it below

but, actually, I just had a look at objdump again. Instead of passing the mangled function name (and depending on a new objdump feature), can we not simply pass the start and end address?

e.g.:
```
objdump -S --start-address=0xdd0 --stop-address=0xe3c tests/test-clients/cpp-inlining/cpp-inlining
```

and isn't the above already covered by SymbolCacheEntry::offset and ::size? 

Meaning: can't we keep the code closer to the current status quo - most notably we really don't need to keep the mangled name anywhere imo.

Reply:
No, we can not. This does not cover zero size symbols. To compute stop-address value we should know function's size.

Reply:
Can you please explain the issue you are seeing with zero size symbols? If it's size is zero then by definition it cannot have any contents that one show - meaning a disassembler is meaningless for it anyways?

Please let's first discuss the above, as it is not clear to me at all - if the outcome is that we really need the demangled names after all, this needs to be put into a comment somewhere. And if we really need it, then please keep the mangled and demangled names next to each other, i.e. into a struct that's stored in `m_dieNameCache` and which gets returned then also from `CuDieRangeMapping::dieName`

Reply:
I mean symbols with zero size similar with encountered in Issue #13 
They have code and disassembler. They are not empty.
They are just dynamically linked and dwfl API is not able to compute their sizes properly.
So we can not compute stop-address for them.

Reply:
Oh? That is the first time I hear that it's a bug in dwfl - please send the details about that to the elfutils mailing list, such that ideally we can find a solution to that on that level.

For me, I would prefer not littering perfparser with workarounds for bugs. Maybe it means we are misusing the (complex) elfutils API. Or it means there's a bug in elfutils that needs to be fixed there.

Either way, this is (also) not directly related to the disassembler patch series itself. So again - can we please keep that one simple for its initial integration and not directly make it super complicated with tons of hacks for some corner cases?

This way, the story encoded in the individual patches also becomes much more clear. And others, like me, can actually follow your reasoning. Right now, I'm totally overwhelmed (as you may realize) by this patch series. I was expecting something much simpler - and would like to see this first.

And remember: always explain in your commit messages *why* something is needed. If there's a bug somewhere like you are saying above that a function that is _not_ empty actually is computed to have zero size, then please put that into the message. Previously when I merged your patch about that, I thought there are functions that have _literally_ no size but still occur for some reason in the symbol table! Now you are saying that is wrong... I would like to learn why!

Reply:
Ok. 
But mangled names refusal mean a lot of reworking for me in 'hotspot' part.
Before reworking them I would like to discuss what should I keep there.
I have entered option --disasm-approach to switch between objdump approach 'by symbol' (objdump -- disassemble='mangledSymName') and 'by address' - it is by start_address and stop_address.
I use 'by symbol' by default.
Do you want me to remove the whole this stuff and keep only 'by addresses' generation method?

Reply:
Keeping track of the mangled sym name makes the code more complex and isn't really needed. It is also not as explicit as passing the start and end addresses directly to `objdump`. And finally it would mean we can use even older versions of `objdump` afaik, which would make this feature more widely available.

So yes, please only keep the `by addresses` method.

Reply:
And generally: we only want one working method, not two methods that do basically the same - the user shouldn't be able to switch between the two methods.